### DISCUSSION TITLE
How Sensitive is the ApiKey?

### DISCUSSION THREAD

**Discussion by Mike-E-angelo** [datalust/seq-tickets#1434] [Help] [answered]
Hello Team Seq,

To start, thank you so much for this product.  It was super easy to stand up without much friction at all.  Very happy with the overall experience. 👍

I am currently getting ready to deploy to Azure, and one of the tasks is configuration.  I currently have the API key stored in plain text in the appsettings.json file, where it will be hosted on the server and of course not available to the public.

My question is: is it OK to store the `apiKey` setting in this file?  Or is the expectation to use Azure Key Vault?  My hunch is that since it's just a key and not a "key-secret" pair that it is OK to store in plain text and not use Azure Key Vault, but wanted to check in here to confirm.

In any case, thank you for the great work out there 👍

**Comment by liammclennan** [comment, marked as answer]
Hi Mike, 
It depends on the permissions assigned to the API key. The API keys you use to send logs from your applications should have only the 'Ingest' permission, that way if someone did get your API key all they can do is write log events.

**Reply by Mike-E-angelo** [reply]
Thank you very much for the reply @liammclennan.  FWIW I was having some confusion on how to use the API key, and was thinking of using a different API key for each application.  However, I decided on using an Enricher instead to delineate my applications, and I will use 1 API key which will be stored in Azure Key Vault. 👍

**Reply by liammclennan** [reply]
That is a reasonable approach. There are some advantages to having an API key per application because it gives you granular visibility and control of ingestion. https://docs.datalust.co/docs/getting-logs-into-seq#api-keys
